packages/twenty-server/src/engine/core-modules/text-to-sql/text-to-sql.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@nestjs/common';
+
+import { ChatOpenAI } from '@langchain/openai';
+import { SqlDatabase } from 'langchain/sql_db';
+import { PromptTemplate } from '@langchain/core/prompts';
+import {
+  RunnablePassthrough,
+  RunnableSequence,
+} from '@langchain/core/runnables';
+import { StringOutputParser } from '@langchain/core/output_parsers';
+
+import { TextToSQLQueryResult } from 'src/engine/core-modules/text-to-sql/dtos/text-to-sql-query-result.dto';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+
+@Injectable()
+export class TextToSQLService {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {}
+
+  async query(
+    workspaceId: string,
+    text: string,
+  ): Promise<TextToSQLQueryResult> {
+    const workspaceSchemaName =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const workspaceDataSource =
```

Comment:
Maybe we should hook into WorkspaceQueryRunner service which already has similar code?
As of now WorkspaceQueryRunner forces query execution through graphql.resolve but maybe refactoring a bit we could rely on `SET search_path` etc and not duplicate that logic? Not 100% sure I understand every single line but it seems similar to what you did here